### ISSUE TITLE
Difficulties when referring to master or develop branch.

### ISSUE THREAD

**Issue by menicosia** [pivotal-cf-experimental/git_repo_changelog#2]
I'm having some trouble prepping for a release. We haven't tagged v0.2.8 of Lattice yet, so I tried to use master or develop branch. When referring to master, I got no results at all. When referring to develop, I got an 'Authorization failure'.

BTW - I haven't ever specified any e-mail addresses since I don't want to restrict to any set of committers - I want all of the commits regardless of who did them. Is that correct?

```
$ bundle exec rake changelog[/Users/pivotal/workspace/lattice,v0.2.7,master,]


$ bundle exec rake changelog[/Users/pivotal/workspace/lattice,v0.2.7,develop,]
rake aborted!
TrackerApi::Error: {:status=>403, :headers=>{"Content-Type"=>"application/json; charset=utf-8", "Status"=>"403 Forbidden", "X-UA-Compatible"=>"IE=Edge,chrome=1", "Cache-Control"=>"no-cache", "X-Request-Id"=>"812a6efac3e4f874a0bc047cb7462b11", "X-Runtime"=>"0.027917", "Date"=>"Tue, 04 Aug 2015 03:15:58 GMT", "X-Rack-Cache"=>"miss", "X-Powered-By"=>"Phusion Passenger 4.0.41", "Server"=>"nginx/1.6.0 + Phusion Passenger 4.0.41"}, :body=>{"code"=>"unauthorized_operation", "kind"=>"error", "error"=>"Authorization failure.", "general_problem"=>"You aren't authorized to access the requested resource.", "possible_fix"=>"Your project permissions are determined on the Project Membership page. If you are receiving this error you may be trying to access the wrong project, or the project API access is disabled, or someone listed as the project's Owner needs to change your membership type."}}
/Users/pivotal/.gem/ruby/2.1.5/gems/tracker_api-0.2.10/lib/tracker_api/client.rb:204:in `rescue in request'
/Users/pivotal/.gem/ruby/2.1.5/gems/tracker_api-0.2.10/lib/tracker_api/client.rb:184:in `request'
/Users/pivotal/.gem/ruby/2.1.5/gems/tracker_api-0.2.10/lib/tracker_api/client.rb:54:in `get'
/Users/pivotal/.gem/ruby/2.1.5/gems/tracker_api-0.2.10/lib/tracker_api/endpoints/story.rb:17:in `get_story'
/Users/pivotal/.gem/ruby/2.1.5/gems/tracker_api-0.2.10/lib/tracker_api/client.rb:140:in `story'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:15:in `block (2 levels) in format'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:14:in `each'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:14:in `block in format'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:12:in `each'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:12:in `format'
/Users/pivotal/workspace/git_repo_changelog/lib/tasks/changelog.rake:15:in `block in <top (required)>'
Faraday::ClientError: the server responded with status 403
/Users/pivotal/.gem/ruby/2.1.5/gems/faraday-0.9.1/lib/faraday/response/raise_error.rb:13:in `on_complete'
/Users/pivotal/.gem/ruby/2.1.5/gems/faraday-0.9.1/lib/faraday/response.rb:9:in `block in call'
/Users/pivotal/.gem/ruby/2.1.5/gems/faraday-0.9.1/lib/faraday/response.rb:57:in `on_complete'
/Users/pivotal/.gem/ruby/2.1.5/gems/faraday-0.9.1/lib/faraday/response.rb:8:in `call'
/Users/pivotal/.gem/ruby/2.1.5/gems/faraday-0.9.1/lib/faraday/rack_builder.rb:139:in `build_response'
/Users/pivotal/.gem/ruby/2.1.5/gems/faraday-0.9.1/lib/faraday/connection.rb:377:in `run_request'
/Users/pivotal/.gem/ruby/2.1.5/gems/faraday-0.9.1/lib/faraday/connection.rb:140:in `get'
/Users/pivotal/.gem/ruby/2.1.5/gems/tracker_api-0.2.10/lib/tracker_api/client.rb:196:in `request'
/Users/pivotal/.gem/ruby/2.1.5/gems/tracker_api-0.2.10/lib/tracker_api/client.rb:54:in `get'
/Users/pivotal/.gem/ruby/2.1.5/gems/tracker_api-0.2.10/lib/tracker_api/endpoints/story.rb:17:in `get_story'
/Users/pivotal/.gem/ruby/2.1.5/gems/tracker_api-0.2.10/lib/tracker_api/client.rb:140:in `story'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:15:in `block (2 levels) in format'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:14:in `each'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:14:in `block in format'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:12:in `each'
/Users/pivotal/workspace/git_repo_changelog/lib/git_repo_changelog/tracker_formatter.rb:12:in `format'
/Users/pivotal/workspace/git_repo_changelog/lib/tasks/changelog.rake:15:in `block in <top (required)>'
Tasks: TOP => changelog
(See full trace by running task with --trace)
```

**Comment by menicosia** [comment]
FYI, this does still work for two tagged versions of Lattice. Here's how I generated the v0.2.7 release notes:

```
$ bundle exec rake changelog[/Users/pivotal/workspace/lattice,v0.2.6,v0.2.7,]
lattice
- User should be able to download v0.2.7 [details](https://www.pivotaltracker.com/story/show/99520558)
- `ltc create` docs incorrectly specify docker ENV directive support [details](https://www.pivotaltracker.com/story/show/99279992)
- Subsequent release past Lattice 0.2.6 should work using Terraform ~~v0.6.0~~ v0.6.1 [details](https://www.pivotaltracker.com/story/show/98493606)
[...]
```

**Comment by mmb** [comment]
This looks like the tracker token you are using cannot access some of stories referenced in the commits. Are you sure the user that owns this token has access to all the trackers?

**Comment by mmb** [comment]
One of the stories referenced is 9876919, which is probably a typo in the commit message and not a story in one of our backlogs. I'll create a story in our backlog to put something like `- story 9876919, failed to fetch title` in the output instead of blowing up the entire process.

**Comment by mmb** [comment]
Created https://www.pivotaltracker.com/story/show/100672338.

**Comment by menicosia** [comment]
Hi @mmb,

It's definitely not an invalid token, I showed the same token working in a comment above.

So, I hope it's due to the typo in the commit message. I think the story that's causing your tool to break might be [#98769190](https://www.pivotaltracker.com/story/show/98769190), but I can't find a reference to '9876919' there.

Re token:
I just double checked, using Shannon's suggestion of quotes to encode the empty string.

This works:

> TRACKER_API_TOKEN=xxx bundle exec rake changelog[/Users/pivotal/workspace/lattice,v0.2.5,v0.2.6,'']

This blows up with a traceback:

> TRACKER_API_TOKEN=xxx bundle exec rake changelog[/Users/pivotal/workspace/lattice,v0.2.7,v0.3.0,'']

We cut the v0.3.0 release of Lattice today, so I'd really like to be able to auto-gen the release notes using your tool.

-- Marco N.

**Comment by mmb** [comment]
This commit was the one I found:

https://github.com/cloudfoundry-incubator/lattice/commit/e4d4ba18f6b4f6e2e1c3429ae14e1b542b4dd5d4

**Comment by crhino** [comment]
Fixed by da14f6d5c0d1fe00959d21d593f3e1c934e19958

**Comment by menicosia** [comment]
Works! Thank you!
